R/read_alevin.R

```diff
@@ -73,28 +82,35 @@ read_alevin <- function(quant_dir,
     sample_id = sample_id)
 
   # if alevin-fry USA and MTX format directly create SCE object with fishpond
-  if(usa_mode) {
-
-    # actually check that files are in usa mode
-    if(meta$usa_mode != TRUE){
-      stop("Output files not in USA mode")
+  if(fry_mode) {
+
+    # only check for usa mode for non-feature data
+    if(!feature_data){
+      # actually check that files are in usa mode
+      if(meta$usa_mode != TRUE){
+        stop("Output files not in USA mode")
+      }
     }
 
     # define assays to include in SCE object based on include_unspliced
     if(include_unspliced){
       assay_formats <- list("counts" = c("S", "A", "U"), "spliced" = c("S", "A"))
       meta$transcript_type <- c("total", "spliced")
-    } else {
+    } else if(!include_unspliced & !feature_data) {
```

Comment:
We already know that include_unspliced is false, since we checked it in the first if...
```suggestion
    } else if(!feature_data) {
```